reframe/core/shell.py

```diff
@@ -115,6 +117,7 @@ class generate_script:
     def __init__(self, filename, *args, **kwargs):
         self._shgen = ShellScriptGenerator(*args, **kwargs)
         self._file = open(filename, 'wt')
+        os.chmod(filename, os.stat(filename).st_mode | stat.S_IEXEC)
```

Comment:
I think better if you set the mode after you write the file in `__exit__()` and I would suggest that the mode is taken as an argument after `filename` with default value set to `None`. If `None` then we do the `+x`  as of here.